zeus/monitor/power.py

```diff
@@ -128,20 +128,24 @@ def __init__(
         if gpu_indices is not None and not gpu_indices:
             raise ValueError("`gpu_indices` must be either `None` or non-empty")
 
-        # Initialize NVML.
-        pynvml.nvmlInit()
+        # Get GPUs
+        gpus = get_gpus(gpu_indices)
 
         # Set up logging.
         self.logger = get_logger(type(self).__name__)
 
-        # Get GPU indices and NVML handles.
-        self.gpu_indices, nvml_gpu_indices = resolve_gpu_indices(gpu_indices)
-        nvml_handles = [pynvml.nvmlDeviceGetHandleByIndex(i) for i in nvml_gpu_indices]
+        # Get GPU indices:
+        # if gpu_indices is not None, we can directly use these indices in the gpu class,
+        # since the class only tracks all CUDA_VISIBLE_DEVICES gpus, or all gpus if CUDA_VISIBLE_DEVICES is not set
+        # if gpu_indices is None, we must track all GPUs visible under CUDA_VISIBLE_DEVICES (which the gpus class keeps track of)
```

Comment:
```suggestion
        # Get GPUs
```
I would say the content of the comment is implicit when the user users the `GPUs` API.